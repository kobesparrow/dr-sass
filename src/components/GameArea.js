import React, { Component } from 'react';

class GameArea extends Component {
  constructor(props) {
    super(props);

    // this.checkAnswer = this.checkAnswer.bind(this);
  }

  checkAnswer = (e) => {
    let answer = e.target.innerText;
    if (answer === this.props.prompt.correctAnswer) {
      this.props.scoreIncrease()
      this.props.populatePrompt(); 
    } else {
      this.props.storeWrongAnswer(this.props.prompt);
      this.props.populatePrompt();
    }
  }

  render() {
    return (
      <section className="gameArea">
        <h3>{this.props.prompt.Question}</h3>
        <div className="answerButtons">
          <button className="answer-one-btn answer-btn" onClick={ this.checkAnswer }>{ this.props.prompt.answers[0] }</button>
          <button className="answer-two-btn answer-btn" onClick={ this.checkAnswer }>{ this.props.prompt.answers[1] }</button>
          <button className="answer-three-btn answer-btn" onClick={ this.checkAnswer }>{ this.props.prompt.answers[2] }</button>
        </div>
      </section>
    )
  }
}

export default GameArea;