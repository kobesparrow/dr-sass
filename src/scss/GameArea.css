.gameArea {
  width: 400px;
  border: 2px solid #082B33;
  border-radius: 25px;
  background: #FFF;
  margin-left: 25%;
  margin-bottom: 300px;
  margin-top: 50px;
}

ul {
  list-style: none;
}

.answerButtons {
  display: flex;
  flex-direction: column
}

button {
  margin-bottom: 25px;
  margin-left: 25px;
  margin-right: 25px;
}