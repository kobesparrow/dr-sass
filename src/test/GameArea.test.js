import React from 'react';
import GameArea from '../components/GameArea';


describe ('GameArea', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <GameArea />
    )
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});