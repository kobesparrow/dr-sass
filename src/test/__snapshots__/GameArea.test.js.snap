// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameArea should match the snapshot with all data passed in 1`] = `
<section
  className="gameArea"
>
  <h3>
    What does the @onceler directive do
  </h3>
  <div
    className="answerButtons"
  >
    <button
      className="answer-one-btn answer-btn"
      onClick={[Function]}
    >
      repeats CSS definitions multiple times in a row
    </button>
    <button
      className="answer-two-btn answer-btn"
      onClick={[Function]}
    >
      sets style based on a condition
    </button>
    <button
      className="answer-three-btn answer-btn"
      onClick={[Function]}
    >
      @onceler is not a Sass directive
    </button>
  </div>
</section>
`;
